WordPress/src/main/java/org/wordpress/android/viewmodel/domains/DomainSuggestionsViewModel.kt

```diff
@@ -0,0 +1,135 @@
+package org.wordpress.android.viewmodel.domains
+
+import android.arch.lifecycle.LiveData
+import android.arch.lifecycle.MutableLiveData
+import android.arch.lifecycle.Transformations
+import android.arch.lifecycle.ViewModel
+import android.text.TextUtils
+import org.greenrobot.eventbus.Subscribe
+import org.greenrobot.eventbus.ThreadMode
+import org.wordpress.android.fluxc.Dispatcher
+import org.wordpress.android.fluxc.generated.SiteActionBuilder
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.network.rest.wpcom.site.DomainSuggestionResponse
+import org.wordpress.android.fluxc.store.SiteStore.OnSuggestedDomains
+import org.wordpress.android.fluxc.store.SiteStore.SuggestDomainsPayload
+import org.wordpress.android.models.networkresource.ListState
+import org.wordpress.android.util.AppLog
+import org.wordpress.android.util.AppLog.T
+import org.wordpress.android.util.helpers.Debouncer
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+import kotlin.properties.Delegates
+
+typealias DomainSuggestionsListState = ListState<DomainSuggestionResponse>
+
+class DomainSuggestionsViewModel @Inject constructor(
+    private val dispatcher: Dispatcher,
+    private val debouncer: Debouncer
+) : ViewModel() {
+    lateinit var site: SiteModel
+    private var isStarted = false
+
+    private val _suggestions = MutableLiveData<DomainSuggestionsListState>()
+    val suggestionsLiveData: LiveData<DomainSuggestionsListState>
+        get() = _suggestions
+
+    private var suggestions: ListState<DomainSuggestionResponse>
+            by Delegates.observable(ListState.Init()) { _, _, new ->
+                _suggestions.postValue(new)
+            }
+
+    private val _selectedSuggestion = MutableLiveData<DomainSuggestionResponse?>()
+    val selectedSuggestion: LiveData<DomainSuggestionResponse?>
+        get() = _selectedSuggestion
+
+    val choseDomainButtonEnabledState: LiveData<Boolean>
+        get() = Transformations.map(_selectedSuggestion) { it is DomainSuggestionResponse }
+
+    private val _selectedPosition = MutableLiveData<Int>()
+    val selectedPosition: LiveData<Int>
+        get() = _selectedPosition
+
+    private var searchQuery: String by Delegates.observable("") { _, oldValue, newValue ->
+        if (newValue != oldValue) {
+            debouncer.debounce(Void::class.java, {
+                fetchSuggestions()
+            }, SEARCH_QUERY_DELAY_MS, TimeUnit.MILLISECONDS)
+        }
+    }
+
+    companion object {
+        private const val SEARCH_QUERY_DELAY_MS = 250L
```

Comment:
It looks like there are various search query delay values throughout the app (e.g. [Pages View Model](https://github.com/wordpress-mobile/WordPress-Android/blob/caee8c89ae1b6729ac8ea65e7137575a78f396d3/WordPress/src/main/java/org/wordpress/android/viewmodel/pages/PagesViewModel.kt#L66), [Site Creation Domains View Model](https://github.com/wordpress-mobile/WordPress-Android/blob/caee8c89ae1b6729ac8ea65e7137575a78f396d3/WordPress/src/main/java/org/wordpress/android/ui/sitecreation/domains/NewSiteCreationDomainsViewModel.kt#L43), [Site Creation Verticals View Model](https://github.com/wordpress-mobile/WordPress-Android/blob/caee8c89ae1b6729ac8ea65e7137575a78f396d3/WordPress/src/main/java/org/wordpress/android/ui/sitecreation/verticals/NewSiteCreationVerticalsViewModel.kt#L45), [Site Creation Domain Adapter](https://github.com/wordpress-mobile/WordPress-Android/blob/caee8c89ae1b6729ac8ea65e7137575a78f396d3/WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SiteCreationDomainAdapter.java#L30), [Username Changer Full Screen Dialog](https://github.com/wordpress-mobile/WordPress-Android/blob/caee8c89ae1b6729ac8ea65e7137575a78f396d3/WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/UsernameChangerFullScreenDialogFragment.java#L75)).  Should we consolidate those into a single constant for a consistent experience when searching?  If the delay should be different based on the API call or what is being searched, should we at least consolidate the domain searching delays (e.g. [Site Creation Domains View Model](https://github.com/wordpress-mobile/WordPress-Android/blob/caee8c89ae1b6729ac8ea65e7137575a78f396d3/WordPress/src/main/java/org/wordpress/android/ui/sitecreation/domains/NewSiteCreationDomainsViewModel.kt#L43), [Site Creation Domain Adapter](https://github.com/wordpress-mobile/WordPress-Android/blob/caee8c89ae1b6729ac8ea65e7137575a78f396d3/WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SiteCreationDomainAdapter.java#L30), and this one)?

Reply:
Wow, there are a lot of places where we use different delays. I don't have enough context for each use case to make a call. I think it's a good candidate for a separate task, so I made an issue here -> #9738